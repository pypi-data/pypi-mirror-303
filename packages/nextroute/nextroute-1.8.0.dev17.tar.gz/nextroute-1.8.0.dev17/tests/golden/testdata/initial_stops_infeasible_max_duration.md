# Initial stop infeasible example (initial_stops_infeasible_max_duration.json)

This example demonstrates the use of the `initial_stops` with optional (not
`fixed`) stops when facing a constraint on the last stop. I.e., the max duration
of the vehicle imposes a last start time on the end stop of the vehicle.

Find some notes about the example below:

- The only vehicle starts with all stops pre-assigned as `initial_stops`.
- Every stop takes 10 minutes to be served and _some_ (fast) driving occurs
  between the stops. This means that only 3 stops can be served within the max
  duration of 35 minutes. All others should get removed in reverse order, but
  should be assigned again since the route is shortest as 5-4-3.
