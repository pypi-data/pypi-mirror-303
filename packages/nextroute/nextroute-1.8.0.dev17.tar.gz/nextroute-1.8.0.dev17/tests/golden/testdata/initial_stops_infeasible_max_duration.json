{
  "defaults": {
    "stops": {
      "unplanned_penalty": 20000,
      "duration": 600
    }
  },
  "stops": [
    {
      "id": "stop1",
      "location": { "lat": 51.9636, "lon": 7.6293 }
    },
    {
      "id": "stop2",
      "location": { "lat": 51.9635, "lon": 7.6439 }
    },
    {
      "id": "stop3",
      "location": { "lat": 51.9635, "lon": 7.6585 }
    },
    {
      "id": "stop4",
      "location": { "lat": 51.9635, "lon": 7.6731 }
    },
    {
      "id": "stop5",
      "location": { "lat": 51.9635, "lon": 7.6877 }
    }
  ],
  "vehicles": [
    {
      "id": "v1",
      "speed": 200,
      "start_time": "2023-01-01T12:00:00Z",
      "start_location": { "lat": 51.9635, "lon": 7.7023 },
      "initial_stops": [
        { "id": "stop1" },
        { "id": "stop2" },
        { "id": "stop3" },
        { "id": "stop4" },
        { "id": "stop5", "fixed": true }
      ],
      "max_duration": 2100
    }
  ]
}
